src/Carousel/Carousel.tsx

```diff
@@ -80,14 +96,19 @@ const Carousel: RsRefForwardingComponent<'div', CarouselProps> = React.forwardRe
 
     const handleTransitionEnd = useCallback(
       (event: React.TransitionEvent<HTMLDivElement>) => {
-        onSlideEnd?.(activeIndex, event);
+        onSlideEnd?.(currentIndex, event);
       },
-      [activeIndex, onSlideEnd]
+      [currentIndex, onSlideEnd]
     );
 
-    // Set a timer for automatic playback.
-    // `autoplay` needs to be cast to boolean type to avoid undefined parameters.
-    const { clear, reset } = useTimeout(handleSlide, autoplayInterval, !!autoplay && count > 1);
+    useEffect(() => {
+      if (inputActiveIndex !== activeIndex) {
+        setInputActiveIndex(activeIndex);
+        if (activeIndex >= 0 && activeIndex !== currentIndex) {
+          handleSlide(activeIndex);
+        }
+      }
+    }, [activeIndex, inputActiveIndex, currentIndex, handleSlide, setInputActiveIndex]);
```

Comment:
After using useControlled to manage activeIndex and defaultActiveIndex, there is no need to update activeIndex through useEffect.
```suggestion
```

Reply:
Changed here https://github.com/rsuite/rsuite/pull/2338/commits/4e28d4c3fe28dd2027e5e5d4599e955499f05996